File: LP1/cpp/main.cpp
#include "DPR.cpp"
#include <list>

using namespace std;

int main()
{
    int stop = 0; // penanda stop
    char pUpdate;
    cout << "Berapa banyak data : ";
    int n, i;
    cin >> n;
    int ID = 1;
    int cariID;
    string nama, bidang, partai;

    // list objek
    list<DPR> llist;
    // iterator yang asalnya dari isi list
    list<DPR>::iterator count = llist.begin();

    // create

    DPR temp; // penampung sementara data

    for (i = 0; i < n; i++)
    {
        cout << "Nama : ";
        cin >> nama;
        cout << "Nama Bidang : ";
        cin >> bidang;
        cout << "Nama Partai : ";
        cin >> partai;

        // memasukan data data atribut menggunakan setter
        temp.setID(ID);
        ID++;
        temp.setNama(nama);
        temp.setBidang(bidang);
        temp.setPartai(partai);

        // masukan ke list
        llist.push_back(temp);
    }

    // menu
    int pilihan = 0;

    // dilakukan sampai user minta udahan
    do
    {
        cout << "MENU" << '\n'
             << "========" << endl;
        cout << "1. Tambah data" << endl;
        cout << "2. Ubah data" << endl;
        cout << "3. Hapus data" << endl;
        cout << "4. Tampil data" << endl;
        cout << "5. Keluar" << endl;

        cin >> pilihan;
        switch (pilihan)
        {

        case 1: // TAMBAH DATA

            cout << "Masukan nama: " << endl;
            cin >> nama;
            cout << "Masukan bidang: " << endl;
            cin >> bidang;
            cout << "Masukan partai: " << endl;
            cin >> partai;

            temp.setID(ID);
            ID++;
            temp.setNama(nama);

            temp.setBidang(bidang);
            temp.setPartai(partai);
            llist.push_back(temp);
            cout << "Berhasil memasukan data baru!";
            break;
        case 2: // UBAH DATA

            cout << "ID yang mau diupdate: ";
            cin >> cariID;
            stop = 0;
            count = llist.begin(); // iterator counter untuk jadi pointer pas searching

            while (count != llist.end() && stop != 1)
            {
                if (count->getID() == cariID) // kalo ketemu IDnya
                {
                    // dikasi pilihan apa yang mau di ubahnya
                    cout << "apa yang mau di ubah: "
                         << '\n';
                    cout << "a. nama" << endl;
                    cout << "b. bidang" << endl;
                    cout << "c. partai" << endl;
                    cin >> pUpdate;

                    switch (pUpdate)
                    {
                    case 'a': // nama
                        cout << "Masukan nama baru: ";
                        cin >> nama;
                        count->setNama(nama);
                        cout << "Update nama berhasil!" << endl;
                        break;
                    case 'b': // bidang
                        cout << "Masukan bidang baru: ";
                        cin >> bidang;
                        count->setBidang(bidang);
                        cout << "Update bidang berhasil!" << endl;
                        break;
                    case 'c': // partai
                        cout << "Masukan partai baru: ";
                        cin >> partai;
                        count->setPartai(partai);
                        cout << "Update partai berhasil!" << endl;
                        break;
                    default:
                        break;
                    }
                    stop = 1;
                }
                else
                {
                    count++;
                }
            }

            break;

        case 3:
            // DELETE
            // mencari id yang mau di hapus terus hapus
            cout << "Masukan id yang ingin di hapus : ";
            cin >> cariID;
            stop = 0;
            count = llist.begin();
            while (count != llist.end() && stop != 1)
            {
                if (count
                        ->getID() == cariID)
                {
                    llist.erase(count);
                    cout << "HAPUS BERHASIL\n";
                    stop = 1;
                }
                else
                {
                    count++;
                }
            }

            break;
        case 4:
            // print
            cout << "===========================\n";
            cout << "| ID |    NAMA    |    BIDANG    |    PARTAI    |";
            cout << endl;
            i = 0;
            for (list<DPR>::iterator it = llist.begin(); it != llist.end(); it++)
            {
                cout << (i + 1) << "." << it->getID()
                     << " | " << it->getNama() << " | " << it->getBidang() << " | " << it->getPartai() << '\n';
                i++;
            }
            break;
        case 5:
            cout << "Keluar mesin.." << endl;
            break;
        }

    } while (pilihan != 5);

    // delete
    return 0;
}

File: LP1/java/Main.java

import java.util.Scanner;

import java.util.ArrayList;

public class Main {

    /**
     * @param args
     */
    public static void main(String[] args) {
        int stop = 0;
        char pUpdate;
        int n, i;
        Scanner sc = new Scanner(System.in);
        ArrayList<DPR> llist = new ArrayList<>();

        int cariId;
        int ID = 0;
        String nama, bidang, partai;

        n = sc.nextInt();
        for (int j = 0; j < n; j++) {
            ID++;
            DPR temp = new DPR();

            System.out.println("Masukan Nama\n");
            nama = sc.next();
            System.out.println("Masukan Bidang\n");
            bidang = sc.next();
            System.out.println("Masukan Partai\n");
            partai = sc.next();

            temp.setID(ID);
            temp.setName(nama);
            temp.setBidang(bidang);
            temp.setPartai(partai);

            llist.add(temp);
        }

        // menu
        int pilihan = 0;
        do {
            System.out.println("\n MENU\n");
            System.out.println("1.Tambah data\n");
            System.out.println("2.Ubah data\n");
            System.out.println("3.Hapus data\n");
            System.out.println("4.Tampil data\n");
            System.out.println("5.Keluar\n");

            pilihan = sc.nextInt();
            switch (pilihan) {
                case 1:
                    DPR temp = new DPR();
                    System.out.println("Masukan Nama\n");
                    nama = sc.next();
                    System.out.println("Masukan Bidang\n");
                    bidang = sc.next();
                    System.out.println("Masukan Partai\n");
                    partai = sc.next();

                    temp.setName(nama);
                    temp.setBidang(bidang);
                    temp.setPartai(partai);

                    llist.add(temp);
                    break;
                case 2:
                    System.out.println("ID yang mau diupdate: ");
                    cariId = sc.nextInt();
                    i = 0;
                    stop = 0;
                    while (i < llist.size() && stop != 1) {
                        if (llist.get(i).getID() == cariId) {
                            System.out.println("apa yang mau diubah\n");
                            System.out.println("a. nama\n");
                            System.out.println("b. bidang\n");
                            System.out.println("c. partai\n");
                            pUpdate = sc.next().charAt(0);
                            switch (pUpdate) {

                                case 'a':
                                    nama = sc.next();
                                    llist.get(i).setName(nama);
                                    break;
                                case 'b':
                                    bidang = sc.next();
                                    llist.get(i).setBidang(bidang);
                                    break;
                                case 'c':
                                    partai = sc.next();
                                    llist.get(i).setPartai(partai);
                                    break;
                                default:
                                    break;
                            }
                            stop = 1;
                        } else {
                            i++;
                        }
                    }
                    break;
                case 3:
                    System.out.println("ID yang mau dihapus: ");
                    cariId = sc.nextInt();
                    stop = 0;
                    i = 0;
                    while (i < llist.size() && stop != 1) {
                        if (llist.get(i).getID() == cariId) {

                            llist.remove(i);
                            System.out.println("Hapus Berhasil\n");
                            stop = 1;
                        } else {
                            i++;
                        }
                    }
                    break;
                case 4:
                    System.out.println("===========================\n");
                    System.out.println("| ID |    NAMA    |    BIDANG    |    PARTAI    |");
                    for (int j = 0; j < llist.size(); j++) {
                        System.out.println(Integer.toString(j + 1) + "." + llist.get(j).getName() + " | "
                                + llist.get(j).getBidang() + " | " + llist.get(j).getPartai() + " |\n");
                    }
                    break;
                case 5:
                    System.out.println("Keluar dari mesin...\n");
                    break;
                default:
                    break;
            }
        } while (pilihan != 5);

    }
}
